src/main/java/org/cryptomator/cryptofs/CryptoPathMapper.java

```diff
@@ -140,14 +141,15 @@ private String getCiphertextFileName(DirIdAndName dirIdAndName) {
 	}
 
 	public void invalidatePathMapping(CryptoPath cleartextPath) {
-		ciphertextDirectories.invalidate(cleartextPath);
+		ciphertextDirectories.synchronous().invalidate(cleartextPath);
 	}
-	
+
 	public void movePathMapping(CryptoPath cleartextSrc, CryptoPath cleartextDst) {
-		CiphertextDirectory cachedValue = ciphertextDirectories.getIfPresent(cleartextSrc);
+		var syncedCache = ciphertextDirectories.synchronous();

```

Comment:
same as above

Reply:
changed in https://github.com/cryptomator/cryptofs/pull/181/commits/1b815373ce749a116d592b34f5d418942c01bf60